src/xdp/rx.c

```diff
@@ -1010,6 +1059,8 @@ XdpRxQueueCreate(
     XdpQueueSyncInitialize(&RxQueue->Sync);
     RxQueue->Binding = Binding;
     RxQueue->Key = Key;
+    RxQueue->InterfaceRxCapabilities.ChecksumOffload = TRUE;
```

Comment:
Why is this hard-coded to TRUE?

Reply:
I believe for TX, the LWF interface is hard-coded to indicate support for checksum offload, but the common XDP code isn't. We still have a notion of "native" XDP drivers that interact directly with the common XDP code instead of going through the NBL code path in the LWF, and those might not support checksum offload.

We have this abstraction for a few reasons:

1. Efficiency: we don't want to descriptor allocate space for extensions that will never be used
2. Correctness: if the interface doesn't support an extension, its contents will be undefined, and not simply ignorable

For RX, assuming XDP zeros the ring contents (which it should, for security) then it is likely the zero values all mean "no checksum" and "layout unknown" but that is not something we want to rely on.

Reply:
Resolved offline.